docs/blog/2023-06-10-encoder-refactors.md

```diff
@@ -0,0 +1,119 @@
+---
+
+Today, we merged a significant change to the low level sensor code that is used to support encoders. In particular,
+this paves the way for completing the work on supporting split peripheral sensors/encoders, and other future sensors
+like pointing devices.
+
+As part of the work, backwards compatibility for existing shields has been retained, but only for a grace period to allow out-of-tree shields to move to the new approach for encoders.
+
+Special thanks to [joelspadin] for the _thorough_ code review and testing throughout the development of the refactor.
+
+## Summary of Changes
+
+The following items have been merged:
+
+1. Split configuration of hardware details, and behavior configuration to allow more flexible functionality of sensors/encoders, in particular linear encoders that lack detents/"clicks" as they rotate.
+2. Support for upstream Zephyr sensor drivers, including the NRFX QDEC driver that can be used on nRF52 based keyboards.
+3. Sensor data handling changes that pave the way for split sensor handling easily.
+
+## Configuration Changes
+
+The major changes to configuration in the devicetree files relates to how the number of steps/triggers for a given encoder are set. In particular, the number of pulses/steps for a given encoder is configured first, allowing ZMK to determine the exact angular degrees of change that is represented by a single pulse on the data lines to that encoder.
+
+Once that angular degrees mapping is completed, now independently there is a configuration setting to control how many triggers of the behavior in the keymap should occur for each full rotation of the sensor. Another way to think of this is "how many degrees of rotation results in a triggering of the sensor behavior in your keymap layer".
+
+Splitting these two parts of the encoder configuration allows greater flexibility, and fine grained control of encoder behavior for linear encoders that don't have fixed detents.
+
+### Old Configuration
+
+Previously, an encoder configuration looked like:
+
+```
+    left_encoder: encoder_left {
+        compatible = "alps,ec11";
+        label = "LEFT_ENCODER";
+        a-gpios = <&pro_micro 21 (GPIO_ACTIVE_HIGH | GPIO_PULL_UP)>;
+        b-gpios = <&pro_micro 20 (GPIO_ACTIVE_HIGH | GPIO_PULL_UP)>;
+        resolution = <4>;
+    };
+```
+
+Here, the `resolution` property was used to indicate how many encoder pulses should trigger the sensor behavior one time. Next, the encoder is selected in the sensors node:
+
+```
+    sensors {
+        compatible = "zmk,keymap-sensors";
+        sensors = <&left_encoder &right_encoder>;
+    };
+```
+
+That was the entirety of the configuration for encoders.
+
+### New Configuration
+
+```
+    left_encoder: encoder_left {
+        compatible = "alps,ec11";
+        label = "LEFT_ENCODER";
+        a-gpios = <&pro_micro 21 (GPIO_ACTIVE_HIGH | GPIO_PULL_UP)>;
+        b-gpios = <&pro_micro 20 (GPIO_ACTIVE_HIGH | GPIO_PULL_UP)>;
+        steps = <20>;
```

Comment:
I think this should be 80 to match the original `resolution = <4>` given the `triggers-per-rotation = <20>` below?

Reply:
Yeah, good call.